compiler/damlc/daml-doc/src/DA/Daml/Doc/Render.hs

```diff
@@ -5,49 +5,112 @@
 
 module DA.Daml.Doc.Render
   ( DocFormat(..)
-  , renderFinish
+  , RenderOptions(..)
+  , RenderMode(..)
+  , renderDocs
+  , renderPage
   , renderSimpleRst
   , renderSimpleMD
-  , renderSimpleHtml
   , renderSimpleHoogle
   , jsonConf
   ) where
 
+import DA.Daml.Doc.Render.Types
 import DA.Daml.Doc.Render.Monoid
 import DA.Daml.Doc.Render.Rst
 import DA.Daml.Doc.Render.Markdown
 import DA.Daml.Doc.Render.Hoogle
 import DA.Daml.Doc.Types
 
+import Data.Maybe
+import Data.List.Extra
+import Data.Foldable
+import System.Directory
+import System.FilePath
+
 import qualified CMarkGFM as GFM
 import qualified Data.Aeson.Encode.Pretty as AP
 import qualified Data.Text as T
-import qualified Data.Text.Lazy as TL
-import qualified Text.Blaze.Html5 as H
-import qualified Text.Blaze.Html.Renderer.Text as H
+import qualified Data.Text.Encoding as T
+import qualified Data.ByteString as BS
+import qualified Data.Map.Strict as Map
 
 
 -- | centralised JSON configuration for pretty-printing
 jsonConf :: AP.Config
 jsonConf = AP.Config (AP.Spaces 2) (AP.keyOrder ["id"]) AP.Generic True
 
+renderDocs :: RenderOptions -> [ModuleDoc] -> IO ()
+renderDocs RenderOptions{..} mods = do
+    let renderModule =
+            case ro_format of
+                Json -> const (renderLine "") -- not implemented (yet?)
+                Hoogle -> const (renderLine "") -- not implemented (yet?)
+                Rst -> renderSimpleRst
+                Markdown -> renderSimpleMD
+                Html -> renderSimpleMD
```

Comment:
Might be worth adding a comment that the conversion to Html is done in postProcessing. I first thought this was a typo :slightly_smiling_face:

Reply:
Maybe this can be made clearer in code by something like
```haskell
let (renderModule, postProcessing) = case ro_format of
      Markdown -> (renderSimpleMD, id)
      Html -> (renderSimpleMD, GFM.commonmarkToHtml [GFM.optUnsafe] [GFM.extTable])
      ...
```

Reply:
Oh, that is more intuitive :-)